### DISCUSSION TITLE
LTSpice_RawWrite for noise plots

### DISCUSSION THREAD

**Discussion by PatriceBar** [nunobrum/PyLTSpice#63] [Q&A] [answered]
Hi,
Thanks for this tool, it is very usefull. I'm struggling to use the LTSpice_RawWrite class to create a new raw file with noise plots inside.
It works well with trans and ac curves but not for noise.
The LTSpice_RawRead class works well with the noise curves, so I'm able to extract the curves from an original raw file and display them with a matplot display but I didn't succeed to create a new raw file. I also tried with the LTSpiceRawWrite.add_traces_from_raw() method but same result.
If someone can help me with that, it would nice.
My aim is to use a batch of noise commands and create a raw file with the compilation of the results get from the different simulations.
Thanks

**Comment by nunobrum** [comment, marked as answer]
Hi @PatriceBar 
I've Just committed a version that allows you to create the Noise Plots. 
The previous version was not able to create them, because it was assuming that files where the first variable is 'frequency' were 'complex'.
This was not fixed alongside with other small issues.
The example below demonstrates how to use it with simulations from different steps. I think you can easily change to merge different simulations. One important thing. It would be best if all simulations have the same axis values. In other words the same .noise command.

```
from PyLTSpice.LTSpice_RawRead import LTSpiceRawRead
from PyLTSpice.LTSpice_RawWrite import LTSpiceRawWrite

LTR = LTSpiceRawRead('Noise.raw')
print(LTR.steps)
LTW = LTSpiceRawWrite()
noise = LTR.get_trace('V(onoise)')
LTW.add_traces_from_raw(LTR, 'V(onoise)', step=0, rename_format="{}_{step}")
LTW.add_traces_from_raw(LTR, 'V(onoise)', step=1, rename_format="{}_{step}")
LTW.save("Noise_stepped.raw")
```

I hope this helps. 
Best Regards,
Nuno

**Reply by PatriceBar** [reply]
Hi @nunobrum,

Thank you so much, it works perfectly. I have used the batch lib to create different raw files and add_traces_from_raw to compile all results in single raw file, it was I wanted to do.
Is it also possible to create a noise raw file using add_traces, without extracting the data from another raw file, like it is done in the example with the time format?
Thanks,
Best regards,
Patrice

**Reply by nunobrum** [reply]
Hi PatriceBar ,

Yes, I think that is possible. Basically it's the same way as we create a time transient raw file.
The only difference is that the first axis is the frequency (not time) and the flags and plotname need to be changed accordingly.
I should some improvements in the sense of making it a bit more usable. I'll signal when this will be done.
Meanwhile here is a non-tested example. 
```
LW = LTSpiceRawWrite()
LW.flag_forward = True
LW.flag_log = True
LW.plot_name = "Noise Spectral Density - (V/Hz½ or A/Hz½)"
axis = Trace('frequency', <array> )
vy = Trace('N001', <array> )
vz = Trace('N002', <array> )
LW.add_trace(axis)
LW.add_trace(vy)
LW.add_trace(vz)
LW.save("teste_w.raw")
```
     
Best Regards,
Nuno
